easybuild/easyconfigs/x/XGBoost/XGBoost-1.6.0-foss-2021a-CUDA-11.3.1.eb

```diff
@@ -0,0 +1,55 @@
+easyblock = 'CMakePythonPackage'
+
+name = 'XGBoost'
+version = '1.6.0'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://github.com/dmlc/xgboost'
+description = """XGBoost is an optimized distributed gradient boosting library designed to be highly efficient,
+ flexible and portable."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+sources = [{
+    'filename': 'xgboost.tar.gz',
+    'git_config': {
+        'url': 'https://github.com/dmlc',
+        'repo_name': 'xgboost',
+        'tag': 'v1.6.0',
+        'recursive': True,
+        'keep_git_dir': True,
+    },
+}]
+
+# Can't add proper SHA256 checksum, because source tarball is created locally after recursive 'git clone'
+checksums = [None]
+
+builddependencies = [
+    ('CMake', '3.20.1'),
+    ('Python', '3.9.5'),
+]
+
+dependencies = [
+    ('CUDA', '11.3.1', '', SYSTEM),
+    ('SciPy-bundle', '2021.05'),
+]
+
+local_cuda_compute_capabilities = '80'
+
+configopts = "-DUSE_CUDA=ON "
+configopts += '-DGPU_COMPUTE_VER=%s' % local_cuda_compute_capabilities
```

Comment:
we shouldn't hardcode the cuda compute capabilities like this.
we have several [templates for different formats](https://docs.easybuild.io/en/latest/version-specific/easyconfig_templates.html?highlight=template#template-values-which-are-defined-dynamically), which are preferable.

Reply:
This was the only way I could get it to work. Believe me, I tried using the correct format but XGBoost didn't cooperate. Having said that, you can give it a try if you want.

Reply:
looking through the build scripts, I think it accepts compute capabilities of the form `-DGPU_COMPUTE_VER="50;70;80;86"`
we don't have a template for that directly, so this might need a simple easyblock to massage the values into the right form -- see the [gromacs easyblock](https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/g/gromacs.py#L197) for an example of this

Reply:
Apparently a template was added recently:

https://github.com/easybuilders/easybuild-framework/pull/4087

I implemented it and it seems to have worked.